src/AutoRest.CSharp.V3/Generation/Writers/JsonCodeWriterExtensions.cs

```diff
@@ -347,6 +353,14 @@ public static void ToDeserializeCall(this CodeWriter writer, JsonValueSerializat
                     }
 
                     writer.AppendRaw(")");
+
+                    // TODO: Turn a GetString() into a char in a safer way
+                    if (frameworkType == typeof(char))
```

Comment:
Can you add a GetChar method to https://github.com/Azure/autorest.csharp/blob/feature/v3/src/assets/Azure.Core.Shared/JsonElementExtensions.cs#L65 and use it higher up